### PR TITLE
Make sure we are taking first dunning user and the rest

### PR DESCRIPTION
**Pull request by cristinamghita**

### CODE REVIEW

File: backend/de.metas.dunning/src/main/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProvider.java [hunk 1]
```diff
@@ -97,6 +98,9 @@ public Optional<DocOutBoundRecipient> provideMailRecipient(@NonNull final DocOut
 		final List<I_C_Invoice> dunnedInvoices = dunningService.retrieveDunnedInvoices(dunningDocId);
 		final int singleCommonInvoiceContactId = CollectionUtils.extractSingleElementOrDefault(dunnedInvoices, I_C_Invoice::getAD_User_ID, -1);
 
+		final I_AD_User dunningContact =  dunningRecord.getC_Dunning_Contact();
```
Comment (adi-stefan) [inline backend/de.metas.dunning/src/main/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProvider.java:101]: please use: de.metas.user.api.impl.UserDAO#getById, which uses a cache

File: backend/de.metas.dunning/src/main/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProvider.java [hunk 2]
```diff
@@ -105,6 +109,43 @@ public Optional<DocOutBoundRecipient> provideMailRecipient(@NonNull final DocOut
...
+			if (Check.isNotBlank(dunningRecipient.getEmailAddress()))
+			{
+				Loggables.addLog("The dunned invoices all have invoiceUser={}, so we take that user as the dunning mail's participant", dunningRecipient);
+				return Optional.of(dunningRecipient);
+			}
+
+			if (Check.isNotBlank(locationEmail))
+			{
+				Loggables.addLog("The dunned invoices all have invoiceUser={} and the location has the email {} so we take that user as the dunning mail's participant, with this email address", dunningRecipient, locationEmail);
+				return Optional.of(dunningRecipient.withEmailAddress(invoiceEmail));
```
Comment (adi-stefan) [inline backend/de.metas.dunning/src/main/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProvider.java:141]:
```suggestion
				return Optional.of(dunningRecipient.withEmailAddress(locationEmail));
```
Shouldn't this be using `locationEmail`, which is what we're checking as not-null? If `invoiceEmail` would be usable, it should have been used already 2 ifs above this one

File: backend/de.metas.dunning/src/test/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProviderTest.java [hunk 3]
```diff
@@ -273,12 +368,16 @@ public void provideMailRecipient_dunned_invoices_without_common_email()
 	}
 
 	@SuppressWarnings("SameParameterValue")
-	private org.compiere.model.I_AD_User createUserRecord(final String eMail)
+	private org.compiere.model.I_AD_User createUserRecord(final String eMail, final boolean... isDunningUser)
```
Comment (adi-stefan) [inline backend/de.metas.dunning/src/test/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProviderTest.java:371]:
```suggestion
	private org.compiere.model.I_AD_User createUserRecord(final String eMail)
	{
		return createUserRecord(eMail, false);
	}	
	private org.compiere.model.I_AD_User createUserRecord(final String eMail, boolean isDunningUser)	
```

File: backend/de.metas.dunning/src/test/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProviderTest.java [hunk 4]
```diff
@@ -273,12 +368,16 @@ public void provideMailRecipient_dunned_invoices_without_common_email()
...
 	{
 		final org.compiere.model.I_AD_User userRecord = newInstance(I_AD_User.class);
 		userRecord.setName("userRecord");
 		userRecord.setEMail(eMail);
 		userRecord.setC_BPartner_ID(bPartnerRecord.getC_BPartner_ID());
+		if (isDunningUser.length > 0)
+		{
+			userRecord.setIsDunningContact(isDunningUser[0]);
+		}
```
Comment (adi-stefan) [inline backend/de.metas.dunning/src/test/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProviderTest.java:380]:
```suggestion
		userRecord.setIsDunningContact(isDunningUser);
```

**Review by adi-stefan — CHANGES_REQUESTED** [review]
Please see above

File: backend/de.metas.dunning/src/main/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProvider.java (hunk 2 above)
Comment (cristinamghita) [reply backend/de.metas.dunning/src/main/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProvider.java:141]: Yes, right!
Was a bug in the code below, that I did not notice, and I copied here as well.
Fixed here and in the bellow if.

**Review by adi-stefan — APPROVED** [review]

**Comment by metas-ts** [comment]
@MergifyIO copy intensive_care_hotfix

**Pull request merged**